src/patternfly/components/CodeSnippet/examples/CodeSnippet.md

```diff
@@ -0,0 +1,146 @@
+---
+id: Code snippet
+section: components
+beta: true
+cssPrefix: pf-c-code-snippet
+---
+
+## Examples
+### Default
+```hbs
+{{#> code-snippet}}
+  {{#> code-snippet-header}}
+    {{#> code-snippet-controls}}
+      {{#> input-group}}
+        {{#> button button--modifier="pf-m-control" button--attribute='aria-label="Copy to clipboard"'}}
+          <i class="fas fa-copy" aria-hidden="true"></i>
+        {{/button}}
+        {{#> button button--modifier="pf-m-control" button--attribute='aria-label="Download code"'}}
+          <i class="fas fa-download"></i>
+        {{/button}}
+        {{#> button button--modifier="pf-m-control" button--attribute='aria-label="Upload code"'}}
+          <i class="fas fa-upload"></i>
+        {{/button}}
+      {{/input-group}}
+    {{/code-snippet-controls}}
+    {{#> code-snippet-tab}}
```

Comment:
I think this will need to be a flex layout as the white space between the icon and text may not always be preserved.